flagd/pkg/service/flag-evaluation/flag_evaluator.go

```diff
@@ -293,21 +301,36 @@ func (s *OldFlagEvaluationService) ResolveObject(
 	return res, err
 }
 
+// mergeContexts combines values from the request context with the values from the config --context-values flag.
+// Request context values have a higher priority.
+func mergeContexts(reqCtx, configFlagsCtx map[string]any) map[string]any {
+	merged := make(map[string]any)
+	for k, v := range configFlagsCtx {
+		merged[k] = v
+	}
+	for k, v := range reqCtx {
+		merged[k] = v
+	}
+	return merged
+}
+
 // resolve is a generic flag resolver
 func resolve[T constraints](ctx context.Context, logger *logger.Logger, resolver resolverSignature[T], flagKey string,
 	evaluationContext *structpb.Struct, resp response[T], metrics telemetry.IMetricsRecorder,
+	configContextValues map[string]any,
```

Comment:
i do not think that we need here another value, we are merging the values with the context already on line https://github.com/open-feature/flagd/pull/1448/files#diff-65826bb36f80a7e210128a2d3493c29be6f6c6f9487feae242eccbbc61c74c4fR75 - can we simply provide the result of the other line as evaluationcontext. like 

```
evalCtx := map[string]any{}
	if e := req.Msg.GetContext(); e != nil {
		evalCtx = e.AsMap()
	}

       // add values to evalCtx here (deleted code from link) and pass only the evalCtx with all the information

	values, err := s.eval.ResolveAllValues(sCtx, reqID, evalCtx)
```
Or access `s.contextValues`  within this method and merge in here. But i do not see the need for an additional `mergeContext`, seems like we are doing this twice now.

Reply:
Sorry, but I'm confused. To me it looks like there're 2 different use cases: [ResolveAll](https://github.com/open-feature/flagd/blob/8c843df7714b1f2d120c5cac8e40c7220cc0c05b/flagd/pkg/service/flag-evaluation/flag_evaluator_v2.go#L52) and the [generic resolve](https://github.com/open-feature/flagd/blob/8c843df7714b1f2d120c5cac8e40c7220cc0c05b/flagd/pkg/service/flag-evaluation/flag_evaluator.go#L297). And these don't seem to be called at the same time. So there're two places where we need to handle it.

Reply:
My bad, you are totally right sorry

Reply:
I am sorry, you are totally right. - I mixed those methods up.